caddyhttp/httpserver/replacer.go

```diff
@@ -224,6 +224,16 @@ func (r *replacer) getSubstitution(key string) string {
 			}
 		}
 	}
+	// search response headers then
+	if key[1] == '<' {
+		want := key[2 : len(key)-1]
+		for key, values := range r.responseRecorder.Header() {
```

Comment:
Note from comment way above in the constructor, NewReplacer: `rr may be nil if it is not available.` -- so do a nil check before line 228 to make sure `r.responseRecorder` isn't nil.

Reply:
@mholt I have added this check and tests now.